### PR TITLE
RPC v0.6.0 documentation

### PR DESCRIPTION
**Pull request by ddoktorski**
Related to #1235 

## Introduced changes

- Minimal changes in docs for `0.19.0` release. It will be revisited before the actual release.

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: docs/migration_guide.rst [hunk 1]
```diff
@@ -5,22 +5,69 @@ Migration guide
...
+Changes in the :class:`~starknet_py.net.account.account.Account`:
+
+.. currentmodule:: starknet_py.net.account.account
+
+- :meth:`~Account.execute_v3` has been added.
+- :meth:`~Account.sign_declare_v3_transaction`, :meth:`~Account.sign_deploy_account_v3_transaction` and :meth:`~Account.sign_invoke_v3_transaction` have been added.
+- :meth:`~Account.sign_declare_transaction`, :meth:`~Account.sign_deploy_account_transaction` and :meth:`~Account.sign_invoke_transaction` have been renamed to :meth:`~Account.sign_declare_v1_transaction`, :meth:`~Account.sign_deploy_account_v1_transaction` and :meth:`~Account.sign_invoke_v1_transaction` respectively.
+
+All new functions with ``v3`` in their name operate similarly to their ``v1`` and ``v2`` counterparts.
+The key difference resides in the parameter acceptance: they don't accept the ``max_fee`` parameter. Instead, they utilize the ``l1_resource_bounds`` parameter.
```
Comment (DelevoXDG) [inline docs/migration_guide.rst:24]: I would rephrase that, so that users know they're required to have STRK tokens to use `v3` function. Example (you can phrase it better):

```suggestion
Unlike their ``v1`` counterparts, ``v3`` transactions pay their fees in Fri (10^-18 STRK). Therefore,  ``max_fee`` parameter, which is typically set in Wei, is not applicable for ``v3`` functions. Instead, ``l1_resource_bounds`` parameter is utilized to limit the Fri amount used.
```

File: docs/migration_guide.rst [hunk 2]
```diff
@@ -5,22 +5,69 @@ Migration guide
...
+    :class:`~starknet_py.contract.Contract` class does not support V3 transactions in the pre-release.
+
+
+0.19.0 Targeted versions
+------------------------
+
+- Starknet - `0.13.0 <https://docs.starknet.io/documentation/starknet_versions/version_notes/#version0.13.0>`_
+- RPC - `0.6.0 <https://github.com/starkware-libs/starknet-specs/releases/tag/v0.6.0>`_
+
 0.19.0 Breaking changes
```
Comment (DelevoXDG) [inline docs/migration_guide.rst:45]: I think `ResourcePrice.price_in_strk` -> `ResourcePrice.price_in_fri` (with later field becoming required) change wasn't mentioned.
Let's make sure we go through #1243 and check if we included all of the breaking changes.

File: docs/migration_guide.rst [hunk 3]
```diff
@@ -5,22 +5,69 @@ Migration guide
...
+
+.. currentmodule:: starknet_py.net.full_node_client
+
+- :meth:`~FullNodeClient.estimate_fee` has a new parameter ``skip_validate``.
+- :meth:`~FullNodeClient.declare` accepts ``transaction`` argument of the type :class:`~starknet_py.net.models.transaction.DeclareV3`
+- :meth:`~FullNodeClient.send_transaction` accepts ``transaction`` argument of the type :class:`~starknet_py.net.models.transaction.InvokeV3`
+- :meth:`~FullNodeClient.deploy_account` accepts ``transaction`` argument of the type :class:`~starknet_py.net.models.transaction.DeployAccountV3`
+
+.. note::
+    :class:`~starknet_py.contract.Contract` class does not support V3 transactions in the pre-release.
```
Comment (THenry14) [inline docs/migration_guide.rst:36]: maybe this should be a warning rather then a note?

File: docs/migration_guide.rst [hunk 4]
```diff
@@ -5,22 +5,71 @@ Migration guide
...
+Changes in the :class:`~starknet_py.net.account.account.Account`:
+
+.. currentmodule:: starknet_py.net.account.account
+
+- :meth:`~Account.execute_v3` has been added.
+- :meth:`~Account.sign_declare_v3_transaction`, :meth:`~Account.sign_deploy_account_v3_transaction` and :meth:`~Account.sign_invoke_v3_transaction` have been added.
+- :meth:`~Account.sign_declare_transaction`, :meth:`~Account.sign_deploy_account_transaction` and :meth:`~Account.sign_invoke_transaction` have been renamed to :meth:`~Account.sign_declare_v1_transaction`, :meth:`~Account.sign_deploy_account_v1_transaction` and :meth:`~Account.sign_invoke_v1_transaction` respectively.
+
+All new functions with ``v3`` in their name operate similarly to their ``v1`` and ``v2`` counterparts.
+Unlike their ``v1`` counterparts, ``v3`` transactions pay their fees in Fri (10^-18 STRK). Therefore,  ``max_fee`` parameter, which is typically set in Wei, is not applicable for ``v3`` functions. Instead, ``l1_resource_bounds`` parameter is utilized to limit the Fri amount used.
```
Comment (DelevoXDG) [inline docs/migration_guide.rst:24]: Sorry I'm bothering again with same thing, but I couldn't help but notice my text implies transactions pay for themselves 😅 
```suggestion
Unlike their ``v1`` counterparts, ``v3`` transaction fees are paid in Fri (10^-18 STRK). Therefore,  ``max_fee`` parameter, which is typically set in Wei, is not applicable for ``v3`` functions. Instead, ``l1_resource_bounds`` parameter is utilized to limit the Fri amount used.
```

Comment (ddoktorski) [reply docs/migration_guide.rst:24]: Missed this on my side. Thank you for being vigilant!

File: docs/guide/deploying_contracts.rst [hunk 5]
```diff
@@ -77,8 +77,8 @@ Declaring Cairo1 contracts
 
 | Starknet 0.11 introduced the ability to declare contracts written in Cairo1!
```
[the remark this answers is not in the document]
Comment (ddoktorski) [reply docs/guide/deploying_contracts.rst:78]: As discussed offline, this should be changed. However, since the current logic at some points in the code allows only Cairo version 0 or 1, we will not update the documentation until it is changed.

**Review by DelevoXDG — APPROVED** [review]
lgtm
